### DISCUSSION TITLE
How to recover blog old blog template?

### DISCUSSION THREAD

**Discussion by gurubyte** [GetPublii/Publii#1515] [Help] [answered]
I had Publii on previous Windows 10 OS. It crashed with blue screen of death but I managed to recover whole Publii directory from it from AppData folder. Now, my blog is still online, with all files on AWS S3 bucket. Im just curious if I can "import" current online blog somehow in new Publii installation on new Windows 10 so I can just continue making posts.

**Comment by candidexmedia** [comment]
Yes, just move your old directory in the new one (hell, your old one might still be there).

See: https://getpublii.com/docs/publii-global-settings.html#fileslocation

**Reply by gurubyte** [reply]
Heh, that wont work in my case because I didn't saved Documents folder on Windows. Like I said AppData folder. Is there other way? I still have blog files in AWS S3 bucket in cloud.

**Reply by candidexmedia** [reply]
Yes, copy your old site folder (including the `input` folder) in AppData and paste it to the sites folder currently set for your new install. You can find where that folder is located in your Settings, as outlined in the link I shared earlier.

![](https://getpublii.com/docs/media/posts/19/files-location.svg)

**Reply by gurubyte** [reply]
I dont have any "input" folder in AppData that is related to Publii. Only related to Microsoft stuff.

**Reply by candidexmedia** [reply]
So when you say:

> I managed to recover whole Publii directory

What does that entail? Because your entire Publii directory should include a dedicated folder for each site, and each site folder should have the following folders:

- input
- output
- preview

You need that input folder to edit and generate new/updated versions of your site. The source files (inclusing the `db.sqlite`) are in there. Try to do another search.

**Reply by candidexmedia** [reply]
For more details: see my previous answer to a similar question: https://github.com/GetPublii/Publii/discussions/1480#discussioncomment-9110425

**Reply by gurubyte** [reply]
![PubliiFolder1](https://github.com/GetPublii/Publii/assets/130367689/1d67fdf2-5223-4a04-b907-d1c42e5fb8a0)
![PubliiFolder2](https://github.com/GetPublii/Publii/assets/130367689/32d8b997-bae5-4df5-86b3-b1a2171f6057)
Those are only 2 saved folder Publii folders I have from old PC. You can see paths in screenshot.
I am not sure that I can recover it with those 2 folders.

Another thing is that I have all website files on S3 Bucket on Amazon, thats where I am hosting blog. And its still active. Files are there. Is it possible to recover it with those files on S3?

**Reply by candidexmedia** [reply]
Gotcha! So you have a directory for the Publii app, but not for your Publii sites.

Unfortunately, you need that entire site folder (including, more importantly, the `input` folder) to generate and publish new edits to your site.

What is sent to your server is the generated site output (HTML, CSS, JS). That will remain online, but the input files that Publii needs to edit and generate your site are different (sqlite database, handlebars files, stylesheets, js, json, theme files, etc.).

**Reply by gurubyte** [reply]
So if understood, Im done? Only way is to make blog again from scratch?

**Reply by candidexmedia** [reply]
If you 1000% cannot find your site files, unfortunately, yes.

If you have a ton of posts, you might have some luck finding a tool which can convert your live site's RSS feed into a Wordpress site archive (WXR), and then [importing the Wordpress site into Publii using Publii's importer](https://getpublii.com/docs/import-wordpress-into-static-html-site.html).

![image](https://github.com/GetPublii/Publii/assets/95003834/4835b84b-30a7-4d62-8d69-241590e6cd08)

**Reply by gurubyte** [reply]
I only have landing page. ABOUT ME page and CONTACT ME page. No posts. I am not sure if it will work like that?

**Reply by dziudek** [reply]
@gurubyte  To be honest - for such simple website I think that manual reproducing the content in Publii will be fastest solution

**Comment by dziudek** [comment, marked as answer]
@gurubyte - unfortunately without having a backup of your source files (input folder), there is no easy way to reproduce a website. @candidexmedia - has a good idea with WP Importer. I can also advice for some better approach - ask ChatGPT or similar tool to write for you a simple crawler which will generate WXR file based on your website sitemap.xml (RSS feed won't contain all subpages). You can specify in your prompt which elements contains content and titles.

**Reply by gurubyte** [reply]
Okay, will try. And what to do then with that WXR file? How to inport it in Publii? What to do after that?

**Reply by dziudek** [reply]
@gurubyte in other thread I have suggested that in my opinion flow with WXR file is unnecessary - the easier way is to reproduce the subpages manually. Flow with WXR makes sense if you have bigger amount of posts

**Reply by gurubyte** [reply]
Yea, okay got it. Lesson learned. Back up everything. :)) thanks for help

**Reply by candidexmedia** [reply]
> And what to do then with that WXR file? How to inport it in Publii? What to do after that?

I agree with what was said about just copy-pasting, but for reference, to answer this question: you would use the WordPress import tool:

![](https://private-user-images.githubusercontent.com/95003834/335775218-4835b84b-30a7-4d62-8d69-241590e6cd08.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTczNTU1MDYsIm5iZiI6MTcxNzM1NTIwNiwicGF0aCI6Ii85NTAwMzgzNC8zMzU3NzUyMTgtNDgzNWI4NGItMzBhNy00ZDYyLThkNjktMjQxNTkwZTZjZDA4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjAyVDE5MDY0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwNGE3ODk1YjM2ZDA2NzNkNDgxYTkyZmFjNDY4ZDBmMmUwY2Q2MjM5NWJmYjYzNDhlNzQ3NDkxYTlmODAzNGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5xdLgcru4RntqCIcwgyerKheoFxV0uEC7kUIateIB-g)

**Comment by jordantrizz** [comment]
I came here after logging into my second workstation, thinking I could install Publii, import my site from GitHub, make changes, and push those changes.

Has there been any thought about shipping the input folder to remote repositories? Specifically, git? Deployment can be handled by whatever service you use, github, gitlab or onedev etc to ensure those files don't reach the deployment server and are publically available?

Do people store their input files in another git repository? Or on Google Drive?

**Reply by candidexmedia** [reply]
My PC is backed up by pCloud (lifetime one-time fee), and I can access all my files from my phone, other PCs, etc. I'd use a cloud service that can be installed as its own "drive" on your workstations.

**Reply by jordantrizz** [reply]
Sure, that works. But if you already have a git repository, wouldn't it make more sense to be hold everything for a site on the git repository?

**Reply by candidexmedia** [reply]
I guess? 🤷🏿‍♀️ I think it's been mentioned on this forum before...

I'm mentioning using a cloud service because that's what the Publii docs suggest for multi-computer use: https://getpublii.com/docs/publii-on-multiple-computers-via-dropbox.html

**Reply by michaelfluegge** [reply]
I sync 2 Macs using Resilio Sync, so I have a real time backup of all my Publii sites (and all my other files, too). So I can work on my blogs with my Mac mini at my desk or at my Macbook in the sun :-) . The only important thing is to quit Puublii on the machine you are not using.   In addition I  sync to a NAS using Resilio Sync.

**Reply by jordantrizz** [reply]
> I guess? 🤷🏿‍♀️ I think it's been mentioned on this forum before...
> 
> I'm mentioning using a cloud service because that's what the Publii docs suggest for multi-computer use: https://getpublii.com/docs/publii-on-multiple-computers-via-dropbox.html

Ah, that makes more sense. That works; use whatever provider you want for Cloud storage. Thanks.

> I sync 2 Macs using Resilio Sync, so I have a real time backup of all my Publii sites (and all my other files, too). So I can work on my blogs with my Mac mini at my desk or at my Macbook in the sun :-) . The only important thing is to quit Puublii on the machine you are not using. In addition I sync to a NAS using Resilio Sync.

I'll probably have to do something similar. I've only got one site, and wanted to keep it all on a git repository and checkout the site when I need to edit it. Thanks.
